test/unit/strainer_unit_test.rb

```diff
@@ -145,4 +145,13 @@ def test_global_filter_clears_cache
     Strainer.global_filter(LateAddedFilter)
     assert_equal 'filtered', Strainer.create(nil).invoke('late_added_filter', 'input')
   end
+
+  def test_add_filter_returns_nil_when_a_filter_module_was_added
+    a = Module.new
+    strainer = Context.new.strainer
+    result1 = strainer.class.add_filter(a)
+    result2 = strainer.class.add_filter(a)
```

Comment:
I don't think the `add_filter` method intended to return anything.  It just happened to use Set#merge on the last line to mutate the set of filter methods.  Perhaps we should be returning `nil` explicitly for public methods so that gem users don't start depending on the return value in cases like this.

What was the undesirable effect of the bug that caused you to want to fix it?  Or did you just happen to notice the bug while reading the code?